src/client/routes/AllFiles/AllFilesView.js

```diff
@@ -17,23 +20,108 @@ export const Title = styled.h1`
 `
 
 class AllFilesView extends Component {
-  handleAddFile = (data) => {
-    const { addFile } = this.props
-    return addFile(data)
+  state = {
+    searchQuery: '',
+    filteredFileList: [],
+  }
+
+  componentDidMount() {
+    const { location } = this.props
+    const queryParams = new URLSearchParams(location.search)
+    const initialQuery = queryParams.get('q') || ''
+
+    this.setState({ searchQuery: initialQuery })
+    this.updateFilteredFileList()
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      this.state.searchQuery !== nextState.searchQuery ||
+      this.state.filteredFileList !== nextState.filteredFileList
+    )
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.searchQuery !== this.state.searchQuery) {
+      this.updateFilteredFileList()
+    }
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.debouncedFiltering)
   }
 
-  renderFiles () {
+  updateFilteredFileList = () => {
+    const { searchQuery } = this.state
     const { files } = this.props
+    const filteredFiles = searchQuery ? files.filter((file) =>
+      file.filename.toLowerCase().startsWith(searchQuery.toLowerCase())
```

Comment:
here I have used `startsWith` method to have a strict comparison to filenames. Incase, we need a more global search support for the names of the file we can instead use `includes`.

`const filteredFiles = searchQuery ? files.filter((file) =>
      file.filename.toLowerCase().includes(searchQuery.toLowerCase())
    ) : files`